cmd/limactl/copy.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"os/user"
+	"strings"
+
+	"github.com/AkihiroSuda/lima/pkg/sshutil"
+	"github.com/AkihiroSuda/lima/pkg/store"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+)
+
+var copyCommand = &cli.Command{
+	Name:        "copy",
+	Aliases:     []string{"cp"},
+	Usage:       "Copy files between host and guest",
+	Description: "Prefix guest filenames with the instance name and a colon.\nExample: limactl copy default:/etc/os-release .",
+	ArgsUsage:   "SOURCE ... TARGET",
+	Action:      copyAction,
+}
+
+func copyAction(clicontext *cli.Context) error {
+	if clicontext.NArg() < 2 {
+		return errors.Errorf("requires at least 2 arguments: SOURCE DEST")
+	}
+	arg0, err := exec.LookPath("scp")
+	if err != nil {
+		return err
+	}
+	u, err := user.Current()
+	if err != nil {
+		return err
+	}
+
+	args, err := sshutil.SCPArgs()
+	if err != nil {
+		return err
+	}
+	args = append(args, "-3", "--")
+	for _, arg := range clicontext.Args().Slice() {
+		path := strings.Split(arg, ":")
+		switch len(path) {
+		case 1:
```

Comment:
Can we require `/` to be included in a host file path?

Because we may want `limactl cp inst1 inst2` to duplicate an instance in future.

Reply:
> Can we require `/` to be included in a host file path?

We could, but it feels slightly wrong to do so. Wouldn't it make more sense to say if none of the arguments contain a colon (i.e. all arguments are host references), then the arguments must be 2 instance names, of which the first one must already exist, and the second one must not?

Because `limactl cp inst1 inst2` would otherwise just be a local copy between 2 files, which doesn't need `limactl cp`, you can use just plain `cp` for that. 😄

Reply:
Ok, I guess we are unlikely going to overload `limactl cp` for copying instances.
For copying instances we can just have `limactl clone inst1 inst2` or maybe `limactl copy-instance inst1 inst2`.

If a user attempted to run `limactl cp inst1 inst2`, we can just print a warning.

Reply:
> Ok, I guess we are unlikely going to overload `limactl cp` for copying instances.
> For copying instances we can just have `limactl clone inst1 inst2` or maybe `limactl copy-instance inst1 inst2`.

Good. I don't really like overloading commands with unrelated functionality, but couldn't think of an "obvious" name for copying an instance. I think `clone` is perfect for this.

> If a user attempted to run `limactl cp inst1 inst2`, we can just print a warning.

Not sure if we need to warn; `limectl cp foo bar` really does the same thing as `cp foo bar` right now, and makes sense to me: if source and dest can be both remote (guest), then why shouldn't they both be allowed to be local (host) as well? And if the source file does not exist, we already get a warning:

```console
$ limactl cp foo bar
cp: foo: No such file or directory
exit status 1
```